### PR TITLE
Update TypeScript.md

### LINKED ISSUE #4122: Wrong type definitions for FastifySchema
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

3.29.1

### Plugin version

6.1.0

### Node.js version

16.15.0

### Operating system

Linux

### Operating system version (i.e. 20.04, 11.3, 10)

Debian 11.3

### Description

typescript throw error when trying to add description or summary to endpoints:
```
fastify.post<{ Body: MyBody}>(
    '/generate',
    {
        schema: {
            description:  'my beautiful description', // <-- type error
            body: myBody,
        },
    },
    handler,
);
```
image

This is supported according to [`@fastify/swagger`](https://github.com/fastify/fastify-swagger):
image

Also, if I'm ignore the error with `// @ts-ignore` it's working correctly and the swagger is valid

### Steps to Reproduce

Almost identical to `@fastify/swagger` example.

1.  save the below code to a file (`a.ts`)
2. run `npx tsc --noEmit a.ts`
```
import fastify from 'fastify';
const server = fastify();
(async () => {

await server.register(require('@fastify/swagger'), {
  routePrefix: '/documentation',
  swagger: {
    info: {
      title: 'Test swagger',
      description: 'Testing the Fastify swagger API',
      version: '0.1.0'
    },
    externalDocs: {
      url: 'https://swagger.io',
      description: 'Find more info here'
    },
    host: 'localhost',
    schemes: ['http'],
    consumes: ['application/json'],
    produces: ['application/json'],
    tags: [
      { name: 'user', description: 'User related end-points' },
      { name: 'code', description: 'Code related end-points' }
    ],
    definitions: {
      User: {
        type: 'object',
        required: ['id', 'email'],
        properties: {
          id: { type: 'string', format: 'uuid' },
          firstName: { type: 'string' },
          lastName: { type: 'string' },
          email: {type: 'string', format: 'email' }
        }
      }
    },
    securityDefinitions: {
      apiKey: {
        type: 'apiKey',
        name: 'apiKey',
        in: 'header'
      }
    }
  },
  uiConfig: {
    docExpansion: 'full',
    deepLinking: false
  },
  uiHooks: {
    onRequest: function (request, reply, next) { next() },
    preHandler: function (request, reply, next) { next() }
  },
  staticCSP: true,
  transformStaticCSP: (header) => header,
  exposeRoute: true
})

server.put('/some-route/:id', {
  schema: {
    description: 'post some data',
    tags: ['user', 'code'],
    summary: 'qwerty',
    params: {
      type: 'object',
      properties: {
        id: {
          type: 'string',
          description: 'user id'
        }
      }
    },
    body: {
      type: 'object',
      properties: {
        hello: { type: 'string' },
        obj: {
          type: 'object',
          properties: {
            some: { type: 'string' }
          }
        }
      }
    },
    response: {
      201: {
        description: 'Successful response',
        type: 'object',
        properties: {
          hello: { type: 'string' }
        }
      },
      default: {
        description: 'Default response',
        type: 'object',
        properties: {
          foo: { type: 'string' }
        }
      }
    },
    security: [
      {
        "apiKey": []
      }
    ]
  }
}, (req, reply) => {})

await server.ready()
server.swagger()

})();

```

### Expected Behavior

no type error

### PR DESCRIPTION
**Pull request by mikicho**
Fix #4122 

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by climba03003 — COMMENTED** [review]
I think we need to elaborate more on this area.

TypeScript can only identify the types that are direct `imported` in code which means that you can use `require` inline with `import` on top.

For example,
```js
import 'plugin' // here will trigger the type augmentation.

fastify.register(require('plugin'))

import plugin from 'plugin' //  here will trigger the type augmentation.
fastify.register(plugin)
```
Or even explicit config on `tsconfig`
```json
{
  "types": ["plugin"] // we force TypeScript to import the types
}
```

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mikicho** [comment]
@climba03003 WDYT?

**Review by climba03003 — APPROVED** [review]
LGTM.

Except the line need to be formatted and break in length 80.

**Review by RafaelGSS — APPROVED** [review]

File: docs/Reference/TypeScript.md [hunk 1]
```diff
@@ -661,6 +661,30 @@ However, there are a couple of suggestions to help improve this experience:
   [npm-check](https://www.npmjs.com/package/npm-check) to verify plugin
   dependencies are being used somewhere in your project.
 
+Note that using `require` won't load the type definitions properly and may
```
Comment (Fdawgs) [inline docs/Reference/TypeScript.md:664]:
```suggestion
Note that using `require` will not load the type definitions properly and may
```

Reference docs are formal, so contractions shouldn't be used.

File: docs/Reference/TypeScript.md [hunk 2]
```diff
@@ -661,6 +661,30 @@ However, there are a couple of suggestions to help improve this experience:
...
+cause type errors.
+TypeScript can only identify the types that are directly imported into code
+which means that you can use require inline with import on top, for example:
```
Comment (Fdawgs) [inline docs/Reference/TypeScript.md:667]:
```suggestion
TypeScript can only identify the types that are directly imported into code,
which means that you can use require inline with import on top. For example:
```

File: docs/Reference/TypeScript.md [hunk 3]
```diff
@@ -661,6 +661,30 @@ However, there are a couple of suggestions to help improve this experience:
...
+```
+
+```typescript
+import plugin from 'plugin' //  here will trigger the type augmentation.
+
+fastify.register(plugin)
+```
+
+Or even explicit config on tsconfig
+```json
```
Comment (Fdawgs) [inline docs/Reference/TypeScript.md:682]:
```suggestion
```jsonc
```

**Review by Fdawgs — CHANGES_REQUESTED** [review]

**Comment by mikicho** [comment]
Thanks! @Fdawgs

**Review by Fdawgs — APPROVED** [review]

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**